framework/src/transfers/MultiAppGeneralFieldTransfer.C

```diff
@@ -743,26 +744,25 @@ MultiAppGeneralFieldTransfer::examineValueConflicts(
     const auto potential_conflict = *conflict_it;
     bool overlap_found = false;
 
-    // Extract info for the potential overlap
+    // Extract info for the potential conflict
     const unsigned int problem_id = std::get<0>(potential_conflict);
     const dof_id_type dof_object_id = std::get<1>(potential_conflict);
     const Point p = std::get<2>(potential_conflict);
     const Real distance = std::get<3>(potential_conflict);
 
-    // Extract variable info
+    // Extract target variable info
     auto & es = getEquationSystem(*_to_problems[problem_id], _displaced_target_mesh);
     System * to_sys = find_sys(es, var_name);
```

Comment:
Would it be hard to just save this `System *`, the same way we save the var name now?  Going from a perfectly good index to a string makes me worry about performance (string comparisons don't belong in inner loops) and worry more about robustness (now that we can support multiple systems it's easy to imagine use cases in which we'd want them to have overlapping variable names...)

Reply:
@friedmud brought the same concern (in the first general field PR) as it's done all over with the names (in fact saving the system does not help, you need to save the variable number as that's retrieved with string comparisons too)

and I profiled and it was irrelevant

overlapping variable names sounds like a bad idea